libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp

```diff
@@ -59,23 +61,54 @@ class MNISTLoader : public DataLoader<uint64_t, T>
     cursor_ = 0;
   }
 
-  std::pair<uint64_t, T> GetAtIndex(uint64_t index) const
+  // One-hot
+  template <typename LabelT = LabelType, typename DataT = T>
+  math::meta::IfIsMathArray<LabelT, std::pair<LabelType, std::vector<T>>> GetAtIndex(uint64_t index)
```

Comment:
should be only one GetAtIndex function with no templating.
Set a member variable on MnistLoader construction to specific whether you're dealing with one-hots or not